copilot-theorem/src/Copilot/Theorem/What4/Translate.hs

```diff
@@ -585,31 +615,41 @@ translateOp1 origExpr sym op xe = case (op, xe) of
           XWord64 e -> XWord64 <$> f e
           _ -> panic [ "Unexpected value in bvOp", show xe ]
 
-        fpOp :: (forall fpp . FPOp1 sym fpp) -> XExpr sym -> IO (XExpr sym)
+        fpOp :: (forall fi . FPOp1 sym fi) -> XExpr sym -> IO (XExpr sym)
         fpOp g xe = case xe of
-          XFloat e -> XFloat <$> g e
-          XDouble e -> XDouble <$> g e
+          XFloat e -> XFloat <$> g WFP.SingleFloatRepr e
+          XDouble e -> XDouble <$> g WFP.DoubleFloatRepr e
           _ -> panic [ "Unexpected value in fpOp", show xe ]
 
         realOp :: RealOp1 sym -> XExpr sym -> IO (XExpr sym)
         realOp h xe = fpOp hf xe
-          where hf :: (forall fpp . FPOp1 sym fpp)
-                hf e = do re <- WI.floatToReal sym e
-                          hre <- h re
-                          WI.realToFloat sym knownRepr fpRM hre
+          where hf :: forall fi . FPOp1 sym fi
+                hf fiRepr e = do re <- WFP.iFloatToReal @_ @fi sym e
+                                 hre <- h re
+                                 WFP.iRealToFloat sym fiRepr fpRM hre
 
         realRecip :: RealOp1 sym
         realRecip e = do one <- WI.realLit sym 1
                          WI.realDiv sym one e
 
+        fpLit :: forall fi.
+                 WFP.FloatInfoRepr fi ->
+                 (forall frac. Fractional frac => frac) ->
+                 IO (WI.SymExpr sym (WFP.SymInterpretedFloatType sym fi))
+        fpLit fiRepr fracLit =
+          case fiRepr of
+            WFP.SingleFloatRepr -> WFP.iFloatLitSingle sym fracLit
+            WFP.DoubleFloatRepr -> WFP.iFloatLitDouble sym fracLit
+            _ -> panic ["Expected single- or double-precision float", show fiRepr]
```

Comment:
This panic is technically new, although I don't know if it's ever reachable due to the way the code is currently structured. As far as I am aware, the `IsInterpretedFloatExprBuilder` only offers `iFloatLit*` methods for `SingleFloat` and `DoubleFloat`.